lib/seahorn/BvOpSem2WideMemManagerMixin.hh

```diff
@@ -169,8 +172,31 @@ public:
   /// \brief returns Expr after setting data.
   PtrTy setFatData(PtrTy p, unsigned SlotIdx, Expr data) override;
 
-  Expr isDereferenceable(PtrTy p, Expr byteSz);
+  Expr isDereferenceable(PtrTy p, Expr byteSz) override;
+
+  MemValTy memsetMetaData(PtrTy ptr, unsigned int len, MemValTy memIn,
+                          uint32_t align, unsigned int val) override;
+
+  MemValTy memsetMetaData(PtrTy ptr, Expr len, MemValTy memIn, uint32_t align,
+                          unsigned int val) override;
+
+  Expr getMetaData(PtrTy ptr, PtrTy memIn, unsigned int byteSz,
+                   uint32_t align) override;
+  unsigned int getMetaDataMemWordSzInBits() override;
 };
 
+// 'HasTracking' is a solution for conditionally compiling in memory tracking
+// code only when needed. It utilizes C++ metaprogramming and LLVM optimization.
+// FIXME: It suffers from the shortcoming that unneeded calls cannot be
```

Comment:
I think if you move the if-then-else statement based on this into template parameter of the function, you will no longer need to define these functions in all memory managers.